decimal.go

```diff
@@ -469,13 +469,30 @@ func (d Decimal) MarshalJSON() ([]byte, error) {
 
 // Scan implements the sql.Scanner interface for database deserialization.
 func (d *Decimal) Scan(value interface{}) error {
-	str, err := unquoteIfQuoted(value)
-	if err != nil {
+	var err error
+	// first try to see if the data is stored in database as a Numeric datatype
+	switch value.(type) {
```

Comment:
So you don't have to do type assertions (value.(float64), etc), you should assign the type switch to t, like in here: https://golang.org/doc/effective_go.html#type_switch